lib/path-handlers.js

```diff
@@ -47,6 +50,54 @@ function get_path(req, res, next) {
     });
 }
 
+
+function get_dir(req, res, next) {
+    var directories = [], juttles = [], relative_path, abs_path;
+
+
+    Promise.try(function() {
+        // massage the heck out of these paths
+        relative_path = req.query.path || '.';
```

Comment:
Probably better to throw an error forcing the client to specify a path than assume anything, I think.